lib/auth.js

```diff
@@ -158,7 +158,12 @@ function obtainToken(accountManagerHostOverride, basicAuthUser, basicAuthPasswor
  */
 function auth(client, clientSecret, user, userPassword, autoRenew, accountManager) {
     // determine oauth flow to use, by default it is resource owner password credentials
-    var flow = { grant : 'password', response_type : 'code' };
+    var flow = { grant : 'client_credentials', response_type : 'token' };
+
+    if ( user || userPassword ) {
```

Comment:
@sftschaerff  

I'm not 100% sure if we switch the clientId/secret combination which was using *password* grant before, does still work for *client_credentials* without changing the related account in the AM.
Did you test that?

Maybe we can just add some sort of deprecation message and leave the flow as it is currently?
So in the end of the file we could place a warning and later on remove the related code?
````
if (flow.grant === 'password') {
  console.warn('some deprecation msg')
}
```

Reply:
As of Gerrit, the client (with the related clientID) should have the role "Account Administrator" and then this should work the same like the old _password_ grant. I have added this requirement to the README section. I haven't tested it.